### PR TITLE
Attempt to test replace before load

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/html/pull/6714.

### CODE REVIEW

**Comment by domenic** [comment]
As I wrote more tests I became more confident that this is just Chrome not behaving as I was told; I think the tests are OK.

Let's do most of the discussion for this in https://github.com/whatwg/html/issues/6579.

Note that the pass condition for all tests currently is "every navigation before load must become a replace" which appears to not match browsers. So we'll need to update that when we settle the issue.

File: html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-load.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  await new Promise(r => { window.onmessage = r; });
+
+  assert_equals(w.history.length, 2, "history.length must increase");
+  assert_equals(w.location.href, absoluteAfterReplacementURL);
+
+  w.history.back();
+
+  t.step_timeout(() => {
+    asssert_equals(w.location.href, absoluteStartURL, "1 second after attempting to go back, it indeed went back");
+  }, 1000);
```
Comment (natechapin) [inline html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-load.html:51]: Here and below: any way to shorten this? 1 second is a long time.

**Review by natechapin — APPROVED** [review]
I know it's what you documented in whatwg/html#6714, but I find the special cases for following hyperlinks and `window.open()` a little jarring. I wonder if that's a necessary permanent behavior or something that we could adjust.

**Comment by domenic** [comment]
> I know it's what you documented in whatwg/html#6714, but I find the special cases for following hyperlinks and window.open() a little jarring.

I suspect window.open() could be changed, as could `aElement.click()`. But user-triggered hyperlinks are probably the way they are for a reason: if you as a user click on a link while a large page (like the HTML spec) is still loading, you probably want your back button to take you back to that large page, and not to skip the large page and go back to whatever you were on previously.

File: html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-pageshow.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  await new Promise(r => { window.onmessage = r; });
+
+  assert_equals(w.history.length, 2, "history.length must increase");
+  assert_equals(w.location.href, absoluteAfterReplacementURL);
+
+  w.history.back();
+
+  t.step_timeout(() => {
+    asssert_equals(w.location.href, absoluteStartURL, "1 second after attempting to go back, it indeed went back");
```
Comment (natechapin) [inline html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-pageshow.html:50]: You've got a couple of 'asssert_equals' typos throughout.

Comment (domenic) [reply html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-pageshow.html:50]: Oh dear, the fact that these tests pass is troubling then. Thanks so much for the careful review.

File: html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-load.html (hunk 1 above)
Comment (domenic) [reply html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-load.html]: Moved to waiting for a posted message

File: html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-pageshow.html (hunk 2 above)
Comment (domenic) [reply html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-pageshow.html]: Fixed

**Review by annevk — APPROVED** [review]

**Comment by foolip** [comment]
@natechapin I've sent you an invite for the wpt reviewers team so that next time your review is enough and you can also merge PRs. (It's write access, just the team isn't named as such.)

**Pull request merged**